x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/extractor/aggregation/AggregationToJsonProcessor.java

```diff
@@ -179,7 +179,7 @@ private void processDateHistogram(Histogram agg) throws IOException {
      * Histograms have either a Double or Long.
      */
     private long toHistogramKeyToEpoch(Object key) {
-        if (key instanceof DateTime) {
+        if (key instanceof DateTime) {//TODO what do we use as key in aggregations
```

Comment:
Is this a question for the ML folk?

Reply:
@dimitris-athanasiou I thought initially to ask ML team, but from what I see now we are missing that migration in elasticsearch. This seems to be a planned work but not started yet.  
Will have to switch to datetime histograms first before finishing ML package

Reply:
That is on the java-time package at the moment and the ZonedDateTime will be returned